spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/autoconfig/AxonTracingSpanAttributeProviderAutoConfiguration.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.springboot.autoconfig;
+
+import org.axonframework.springboot.TracingProperties;
+import org.axonframework.tracing.SpanAttributesProvider;
+import org.axonframework.tracing.SpanFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.List;
+
+/**
+ * Responsible for configuring all registed {@link SpanAttributesProvider} beans to the active {@link SpanFactory}
```

Comment:
Curious: why make a new class for this, as you can argue this is base tracing-config.
Assuming the latter, I'd expect it in the `AxonTracingAutoConfiguration`.

Reply:
One simple reason: Circular dependencies. If we put it in the `AxonTracingAutoConfiguration`, the following happens:

This approach requires the `SpanFactory` bean to be created before `AxonTracingAutoConfiguration` is instantiated (since it's `Autowired` annotation on a method). However, the bean is defined in the `AxonTracingAutoConfiguration` itself, so it gets stuck

Reply:
Can you solve that by exposing a `ConfigurerModule` bean instead?

Reply:
Yes, I wasn't aware of this. Updated the PR.